edfi_api_client/edfi_endpoint.py

```diff
@@ -1,711 +1,533 @@
         Rows are returned as a generator.
 
+        :param params:
         :param page_size:
-        :param retry_on_failure:
-        :param max_retries:
-        :param max_wait:
         :return:
         """
-        if 'step_change_version' in kwargs or 'change_version_step_size' in kwargs or 'reverse_paging' in kwargs:
-            raise KeyError(
-                "Change versions are not implemented in composites!\n"
-                "Remove `step_change_version`, `change_version_step_size`, and/or `reverse_paging` from arguments."
-            )
+        if kwargs.get('step_change_version'):
+            logging.warning("Change versions are not implemented in composites! Change version stepping arguments are ignored.")
+
+        logging.info(f"[Paged Get {self.component}] Endpoint: {self.url}")
+        logging.info(f"[Paged Get {self.component}] Pagination Method: Offset Pagination")
 
         # Reset pagination parameters
-        paged_params = self.params.copy()
+        paged_params = (params or self.params).copy()
         paged_params.init_page_by_offset(page_size)
 
         # Begin pagination-loop
-        self.client.verbose_log(f"[Paged Get Composite] Endpoint  : {self.url}")
-
         while True:
-            self.client.verbose_log(f"[Paged Get Composite] Parameters: {paged_params}")
 
-            if retry_on_failure:
-                res = self._get_response_with_exponential_backoff(
-                    self.url, params=paged_params,
-                    max_retries=max_retries, max_wait=max_wait
-                )
-            else:
-                res = self._get_response(self.url, params=paged_params)
+            ### GET from the API and yield the resulting JSON payload
+            logging.info(f"[Paged Get {self.component}] Parameters: {paged_params}")
+            res = self.client.session.get_response(self.url, params=paged_params, **kwargs)
 
-            # If no rows are returned, end pagination.
-            if len(res.json()) == 0:
-                self.client.verbose_log(f"[Paged Get Composite] @ Retrieved zero rows. Ending pagination.")
-                break
+            # If rows have been returned, there may be more to ingest.
+            if res.json():
+                logging.info(f"[Paged Get {self.component}] Retrieved {len(res.json())} rows.")
+                yield res.json()
 
-            # Otherwise, paginate offset.
+                logging.info(f"    @ Paginating offset...")
+                paged_params.page_by_offset(page_size)
+
+            # If no rows are returned, end pagination.
             else:
-                self.client.verbose_log(f"[Paged Get Composite] @ Retrieved {len(res.json())} rows. Paging offset...")
-                yield res.json()
-                paged_params.page_by_offset()
+                logging.info(f"[Paged Get {self.component}] @ Retrieved zero rows. Ending pagination.")
+                break
 
+    def post(self, *args, **kwargs):
```

Comment:
A note: we've deprecated these methods in the composite without defining them in the main endpoint. This is okay for now.